### PR TITLE
Fix janitor on demand report api bug

### PR DESCRIPTION
**Pull request by daliwei**
cc @pierre for review on the bug fix around c64ba08

### CODE REVIEW

File: lib/cybersource/cyber_source_on_demand.rb [hunk 1]
```diff
@@ -94,6 +94,10 @@ def empty?
           @response.params['merchantReferenceCode'].nil?
         end
 
+        def request_id
+          @response.params['requestID']
```
Comment (pierre) [inline lib/cybersource/cyber_source_on_demand.rb]: Confused: the report is retrieved by querying by order_id and will contain all transactions for that payment (auth, capture(s), refund(s), etc.).

To which transaction will that `requestID` correspond?

File: lib/cybersource/api.rb [hunk 2]
```diff
@@ -145,6 +145,15 @@ def get_payment_info(kb_account_id, kb_payment_id, properties, context)
...
           end
 
+          response = CybersourceResponse.find_by(:id => cybersource_response_id)
+          if response.nil?
+            logger.warn("Unable to fix UNDEFINED kb_transaction_id='#{transaction_info_plugin.kb_transaction_payment_id}' (CyberSource response='#{cybersource_response_id}' not found)")
+            next
+          end
+
+          request_id = response.params_request_id
+          existing_request_ids = CybersourceResponse.where(:kb_payment_id => kb_payment_id).map{|r| r.params_request_id}.compact
```
Comment (daliwei) [inline lib/cybersource/api.rb]: Addressed in 0fef21b

File: lib/cybersource/cyber_source_on_demand.rb (hunk 1 above)
Comment (daliwei) [inline lib/cybersource/cyber_source_on_demand.rb]: The request id should be the last transaction id that occurred from CYBS' POV (see [here](https://github.com/killbill/killbill-cybersource-plugin/blob/e9a3d8a43dd15bb9d4fb180e3a4c010934e38fc0/lib/cybersource/cyber_source_on_demand.rb#L156))

**Comment by pierre** [comment]
It looks like tests aren't passing anymore if the on_demand API is disabled. Could you take a look?

Tracked by https://github.com/killbill/killbill-cybersource-plugin/issues/16.

**Comment by daliwei** [comment]
Sure. Will take a look.

**Pull request merged**